airqo-platform/AirQo-api
# Forecast job refactoring

**_WHAT DOES THIS PR DO?_**
- adds authentication header to api calls 
**_WHAT ISSUES ARE RELATED TO THIS PR?_**

**_HOW DO I TEST OUT THIS PR?_**
   - Fetch this branch and Checkout the branch locally
   - Change directory to Predict/jobs/train/
   - Create a python virtual environment (python3 -m venv env)
   - Install the necessary packages (pip install -r requirements.txt). 
   - Set the environment in the .env file to staging and run python main.py.

- If you have docker, simply run docker build -t <image_name> --target=staging .. Wait for the build process to complete and then run docker run -it <image_name>

- NB:
- .env and service account key is available from the secret manager.
- can check whether job has ran successfuly by observing the results on console and checking the model tracking mlflow interface 
**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**
